engine/runtime-language-arrow/src/main/java/org/enso/interpreter/arrow/runtime/ByteBufferDirect.java

```diff
@@ -201,12 +201,15 @@ private void setValidityBitmap(int index0, int unitSize) {
     var index = index0 / unitSize;
     var bufferIndex = index >> 3;
     var slot = bitmapBuffer.get(bufferIndex);
-    var byteIndex = index & ~(1 << 3);
+    var byteIndex = index & byteMask;
+
     var mask = 1 << byteIndex;
     var updated = (slot | mask);
     bitmapBuffer.put(bufferIndex, (byte) (updated));
   }
 
+  private static final int byteMask = ~(~(1 << 3) + 1); // 7
```

Comment:
Isn't  `0x07` a better way to write a mask? At least that's what I used to do in my childish assembly years. Or is there some logic behind the constant `7` that this double negation and a shift formula is supposed to hint?

```java
jshell> 1 << 3
$1 ==> 8

jshell> ~ (1 << 3)
$2 ==> -9

jshell> ~ (1 << 3) + 1
$3 ==> -8

jshell> ~ (~ (1 << 3) + 1)
$4 ==> 7

jshell> 0x07
$5 ==> 7
```
Thanks for the fix.

Reply:
With your fix (and eliminatation of `null` values from the array), the benchmark got faster. We used to be twenty times slower than plain Java, now it is much better:
```
sbt:std-benchmarks> benchOnly Plus_Fitting
[info] Arrow_Arithmetic_1000000.Plus_Fitting   avgt    3  21.416 ± 99.506  ms/op
[info] Column_Arithmetic_1000000.Plus_Fitting  avgt    3   1.543 ±  0.260  ms/op
```
and I also managed to get an IGV graph from the run. The morning looks great!

![ArrowOperationPlus](https://github.com/enso-org/enso/assets/26887752/69a65ee4-7466-4859-acf5-c7648ad0d3bb)